cmd/thanos/sidecar.go

```diff
@@ -281,10 +291,13 @@ func runSidecar(
 				minTime, _, err := s.Timestamps()
 				if err != nil {
 					level.Warn(logger).Log("msg", "reading timestamps failed", "err", err)
-				} else {
-					m.UpdateTimestamps(minTime, math.MaxInt64)
+					return nil
+				}
+
+				m.UpdateTimestamps(minTime, math.MaxInt64)
+				mint, maxt := m.Timestamps()
 
-					mint, maxt := m.Timestamps()
+				if peer != nil {
```

Comment:
I think we don't need this as we have noop peer in the gossip, and this will be removed in https://github.com/improbable-eng/thanos/pull/1008 so less merge issues for me :D